File: Class Representation/Dependency/Typescript/Dependency.ts
class Documents {
  private content: string;

  constructor(content: string) {
    this.content = content;
  }

  getContent(): string {
    return this.content;
  }
}

class Printer {
  printing(docs: Documents): void {
    console.log(`Printing ${docs.getContent()}`);
  }
}

function main1(): void {
  const doc = new Documents("Hello World");
  const printer = new Printer();

  printer.printing(doc);
}


main1();

File: Class Representation/Dependency/C++/Dependency.cpp
#include <iostream>
using namespace std;

class Document
{
    string content;

public:
    Document(string content)
    {
        this->content = content;
    }

    string getContent()
    {
        return this->content;
    }
};

class Printer
{
public:
    void printing(Document &docs)
    {
        cout << "Printing " << docs.getContent() << endl;
    }
};

int main()
{

    Document docs("Hello world");

    Printer printer;

    printer.printing(docs);

    return 0;
}